(#20) added docs for region (#24)
sakuli/docs: content/writing_tests/screenshot.md

    try {
        await _navigateTo(url);
        await env.setSimilarity(0.8);
        await screen
            .find("source_egg.png")
            .mouseMove()
            .mouseDown(MouseButton.LEFT)
            .setX(0)
            .setY(0)
            .mouseMove()
            .move(200, 700)
            .mouseMove();
        await screen.find("target_pan.png")
            .mouseMove()
            .mouseUp(MouseButton.LEFT);
        await _wait(3000);
    } catch (e) {
        await testCase.handleException(e);
    } finally {
        testCase.saveResult();
    }
})();
{{< /highlight >}}

## How to Control the Mouse in a Sakuli Test

Before using a `region` it is best to instantiate a `region` over the entire display, to use it multiple times shown in

> the examples above.

Reviewer: This sentence somehow feels missplaced.

The heading says _"How to Control the Mouse..."_ but does not mention it at all in the first sentence.
Maybe something like

"Sakuli provides two ways to steer mouse movement. Either via coordinates, or via screenshot data. In order to search for image data on your screen you have to instantiate a `Region` spanning the entire display to access it's image content."

will outline the big picture of mouse movement while explaining why a `Region` is required.

### Moving the mouse to a screenshot

To find a screenshot on the screen, you can use `find` or `waitForImage`. With `waitForImage` you can specify a timeout
in which Sakuli continuously searches for the screenshot. This is useful, when e.g. starting a program like excel.

{{< highlight typescript >}}

> await screen.find("search_this_screenshot.jpg")

Reviewer: We should add a line

```js
const screen = new Region();
```

to all samples for consistency.

    .mouseMove();
    
//searches the screen for 5 seconds for the excel homescreen
await screen.waitForImage("excel_homescreen.jpg", 5000)
    .mouseMove();
{{< /highlight >}} 

> `find` and `waitForImage` returns the found screenshot as a region, so chaining as follows might not work, if the second

Reviewer: ```suggestion
`find` and `waitForImage` will return a `Region` for matched screenshots, so chaining as follows might not work, if the second
```

> screenshot is not in the region of the first one.

Reviewer: ```suggestion
screenshot is not within the `Region` of the first one.

{{< highlight typescript >}}
await screen.find("google_search.png")
    .mouseMove()
    .find("browser_address_bar.jpg")
    .mouseMove();
{{< /highlight >}}

### Moving the mouse relative to current position

> A method to change the position is to use the `move()` method. This allows us to shift the used region, e.g.

Reviewer: ```suggestion
One way to change a `Region`'s position is to use the `move(x, y)` method. This allows us to shift a region in `x` and / or `y` direction, e.g.
```

{{< highlight typescript >}}
await screen.find("login_mask.png")
    .mouseMove()
    .click()
    .type("MyUsername")
    //shifts the region 50px down to e.g. a password field
    .move(0,50)
    //moves the mouse according to the shifted region
    .mouseMove()
    .click()
    .type("MyPassword")
    //shifts the region 50px down and 50px to the right and moves the mouse to e.g. the login button
    .move(50,50)
    .mouseMove()
    .click();
})();
{{< /highlight >}}

### Moving the Mouse to an Absolute Position

> As mentioned before, you need to create a region with a width and heigth of 1 and your x/y coordinates.

Reviewer: Where did we mention to create a region with a width and heigth of 1 and our x/y coordinates? 😃

Author: I think i removed the part where it was mentioned and forgot to update this :see_no_evil:

    .mouseMove();
{{< /highlight >}} 

`find` and `waitForImage` will return a `Region` for matched screenshots, so chaining as follows might not work, if the
second screenshot is not within the `Region` of the first one.
{{< highlight typescript >}}
const screen = new Region();
await screen.find("google_search.png")
    .mouseMove()
    .find("browser_address_bar.jpg")
    .mouseMove();
{{< /highlight >}}

To get around this problem, you can use `left(range)`, `right(range)`, `above(range)` or `below(range)`, which returns a
new `Region`. `left(range)` return a `Region` that is left of the current region's border with a width of _range_, so it
does not include the current Region. `right`, `below` and `above` behaves equivalently.

{{< highlight typescript >}}
const screen = new Region();
await new Region(500, 500, 100, 100).left(200)
//this returns a new region with (300/500/200/100)
{{< /highlight >}}

You can also manipulate the current `Region` with `setX(number)`, `setY(number)`, `setH(number)`, `setW(number)` or
`grow(range)`. With `grow` you can expand the current region with _range_ px in all four directions.
{{< highlight typescript >}}
await screen = new Region();

> await new Region(500, 500, 100, 100).grow(100);

Reviewer: Number are a bit off here :)
